torch/_dynamo/variables/base.py

```diff
@@ -185,9 +185,55 @@ def const_getattr(self, tx, name: str) -> Any:
         """getattr(self, name) returning a python constant"""
         raise NotImplementedError()
 
+    def dynamic_getattr(self, tx, name):
+        if not self.source:
+            raise NotImplementedError()
+
+        if not is_from_local_source(self.source):
+            raise NotImplementedError()
+
+        # For local source, we associate the real value. We use this real value
+        # for implementing getattr fallthrough on the variable tracker base class.
+
+        # Note - this scope construction is mirrored in guards
+        # A subsequent PR will introduce a util.
+        scope = {"L": tx.output.local_scope, "G": tx.output.global_scope}
+        try:
+            # We raise in case we get a typerror bug w/ SuperSource.
+            # SuperSource has bugs in it atm, and can produce code like
+            # eval("super(L['mod'].model.model.encoder.embed_positions.forward__class__,
+            # L['mod'].model.model.encoder.embed_positions)", scope)
+            # Which is incorrect, and violates the invariant that all sources should be eval()-able against the scope.
+            _input_associated_real_value = eval(self.source.name(), scope)
+        except TypeError:
+            raise NotImplementedError()
+        if _input_associated_real_value is None:
+            raise NotImplementedError()
+
+        from .builder import VariableBuilder
+
+        real_value = getattr(_input_associated_real_value, name)
+        if callable(real_value):
+            # Callables have more nuanced handling, and we should let the existing system delegate here.
+            # Raising was past behavior and so should always be sound to fall back.
+            # Note - at a certain point we may want to handle
+            raise NotImplementedError()
+
+        return VariableBuilder(tx, AttrSource(self.source, name))(
+            real_value
+        ).add_options(self)
```

Comment:
We need some sort of guard that the attribute exists in a future call.

Reply:
Yeah, I think a hasattr guard would not be amiss! Good point.

We have that in `HASATTR`